File: Debug/C++/3.cpp
// Find the maximum in the array without using max function.

#include <iostream>
#include <vector>
using namespace std;

int findMax(vector<int>& arr) {
    int maxElement = arr[0];
    for (int i = 1; i < arr.size(); i++) {
        if (arr[i] < maxElement) {
            maxElement = arr[i];
        }
    }
    return maxElement;
}

int main() {
    vector<int> array;
    cout << "Maximum element in the array: " << findMax(array) << endl;

    vector<int> array2 = {3, 7, 2, 9, 5, 8};
    cout << "Maximum element in the array: " << findMax(array2) << endl;

    vector<int> array3 = {-10, -5, -3, -7, -2};
    cout << "Maximum element in the array: " << findMax(array3) << endl;

    return 0;
}

File: Debug/Python/3.py
# Find the maximum in the array without using max function.

def find_max(arr):    
    max_element = arr[0] 
    for num in arr[1:]:
        if num < max_element:
            max_element = num
        return max_element

array = []
print("Maximum element in the array:", find_max(array))
array2 = [3, 7, 2, 9, 5, 8]
print("Maximum element in the array:", find_max(array2))  # Expected output: 9
array3 = [-10, -5, -3, -7, -2]
print("Maximum element in the array:", find_max(array3))  # Expected output: -2
